foresight/utils/handler_utils.py

```diff
@@ -0,0 +1,187 @@
+            test_wrapper_utils = TestWrapperUtils.get_instance()
+            context = test_wrapper_utils.create_test_suite_execution_context(test_suite_id)
+            ExecutionContextManager.set(context)
+            test_wrapper_utils.change_app_info(app_info)
+            TestRunnerSupport.set_test_suite_application_info(app_info)
+            TestRunnerSupport.set_test_suite_execution_context(context)
+            test_wrapper_utils.before_test_process(context)
+
+
+    @classmethod
+    def start_before_all_span(cls, app_info, span_tags= None):
+        return cls.create_span(cls.TEST_BEFORE_ALL_OPERATION_NAME, app_info, span_tags)
+
+
+    @classmethod
+    def finish_before_all_span(cls, scope):
+        cls.finish_span(scope, TestRunnerTags.TEST_BEFORE_ALL_DURATION)
+
+
+    @classmethod
+    def start_after_all_span(cls, app_info, span_tags):
+        """after all executed after test cases. That is why context should be getting from TestRunnerSupport and
+        set into ExecutionContextManager.
+        """
+        context = TestRunnerSupport.test_suite_execution_context
+        ExecutionContextManager.set(context)
+        return cls.create_span(cls.TEST_AFTER_ALL_OPERATION_NAME, app_info, span_tags)
+
+
+    @classmethod
+    def finish_after_all_span(cls, scope):
+        cls.finish_span(scope, TestRunnerTags.TEST_AFTER_ALL_DURATION)
+
+
+    @staticmethod
+    def finish_test_suite_span():
+        test_wrapper_utils = TestWrapperUtils.get_instance()
+        context = ExecutionContextManager.get()
+        context.completed = True
+        test_wrapper_utils.after_test_process(context)
+        TestRunnerSupport.clear_state()
+
+
+    @classmethod
+    def start_test_span(cls, name, test_suite_name, test_case_id, app_info):
+        test_wrapper_utils = TestWrapperUtils.get_instance()
+        test_wrapper_utils.change_app_info(app_info)
+        current_context = ExecutionContextManager.get()
+        parent_transaction_id = current_context.invocation_data.get("transactionId")
+        context = test_wrapper_utils.create_test_case_execution_context(name, test_suite_name, test_case_id, app_info, parent_transaction_id)   
+        ExecutionContextManager.set(context)
+        test_wrapper_utils.before_test_process(context)
+
+
+    @classmethod
+    def start_before_each_span(cls, app_info, span_tags=None):
+        return cls.create_span(cls.TEST_BEFORE_EACH_OPERATION_NAME, app_info, span_tags)
+
+
+    @classmethod
+    def finish_before_each_span(cls, scope):
+        cls.finish_span(scope, TestRunnerTags.TEST_BEFORE_EACH_DURATION)
+
+
+    @classmethod
+    def start_after_each_span(cls, app_info, span_tags=None):
+        return cls.create_span(cls.TEST_AFTER_EACH_OPERATION_NAME, app_info, span_tags)
+
+
+    @classmethod
+    def finish_after_each_span(cls, scope):
+        cls.finish_span(scope, TestRunnerTags.TEST_AFTER_EACH_DURATION)
+
+
+    @staticmethod
+    def finish_test_span():
```

Comment:
where we close test span?

Reply:
It is closed into plugin.py when whole process for test case finishes.